core/vm/jump_table.go

```diff
@@ -22,152 +22,834 @@ import (
-	jumpTable[PUSH4] = jumpPtr{makePush(4, big.NewInt(4)), true}
-	jumpTable[PUSH5] = jumpPtr{makePush(5, big.NewInt(5)), true}
-	jumpTable[PUSH6] = jumpPtr{makePush(6, big.NewInt(6)), true}
-	jumpTable[PUSH7] = jumpPtr{makePush(7, big.NewInt(7)), true}
-	jumpTable[PUSH8] = jumpPtr{makePush(8, big.NewInt(8)), true}
-	jumpTable[PUSH9] = jumpPtr{makePush(9, big.NewInt(9)), true}
-	jumpTable[PUSH10] = jumpPtr{makePush(10, big.NewInt(10)), true}
-	jumpTable[PUSH11] = jumpPtr{makePush(11, big.NewInt(11)), true}
-	jumpTable[PUSH12] = jumpPtr{makePush(12, big.NewInt(12)), true}
-	jumpTable[PUSH13] = jumpPtr{makePush(13, big.NewInt(13)), true}
-	jumpTable[PUSH14] = jumpPtr{makePush(14, big.NewInt(14)), true}
-	jumpTable[PUSH15] = jumpPtr{makePush(15, big.NewInt(15)), true}
-	jumpTable[PUSH16] = jumpPtr{makePush(16, big.NewInt(16)), true}
-	jumpTable[PUSH17] = jumpPtr{makePush(17, big.NewInt(17)), true}
-	jumpTable[PUSH18] = jumpPtr{makePush(18, big.NewInt(18)), true}
-	jumpTable[PUSH19] = jumpPtr{makePush(19, big.NewInt(19)), true}
-	jumpTable[PUSH20] = jumpPtr{makePush(20, big.NewInt(20)), true}
-	jumpTable[PUSH21] = jumpPtr{makePush(21, big.NewInt(21)), true}
-	jumpTable[PUSH22] = jumpPtr{makePush(22, big.NewInt(22)), true}
-	jumpTable[PUSH23] = jumpPtr{makePush(23, big.NewInt(23)), true}
-	jumpTable[PUSH24] = jumpPtr{makePush(24, big.NewInt(24)), true}
-	jumpTable[PUSH25] = jumpPtr{makePush(25, big.NewInt(25)), true}
-	jumpTable[PUSH26] = jumpPtr{makePush(26, big.NewInt(26)), true}
-	jumpTable[PUSH27] = jumpPtr{makePush(27, big.NewInt(27)), true}
-	jumpTable[PUSH28] = jumpPtr{makePush(28, big.NewInt(28)), true}
-	jumpTable[PUSH29] = jumpPtr{makePush(29, big.NewInt(29)), true}
-	jumpTable[PUSH30] = jumpPtr{makePush(30, big.NewInt(30)), true}
-	jumpTable[PUSH31] = jumpPtr{makePush(31, big.NewInt(31)), true}
-	jumpTable[PUSH32] = jumpPtr{makePush(32, big.NewInt(32)), true}
-	jumpTable[DUP1] = jumpPtr{makeDup(1), true}
-	jumpTable[DUP2] = jumpPtr{makeDup(2), true}
-	jumpTable[DUP3] = jumpPtr{makeDup(3), true}
-	jumpTable[DUP4] = jumpPtr{makeDup(4), true}
-	jumpTable[DUP5] = jumpPtr{makeDup(5), true}
-	jumpTable[DUP6] = jumpPtr{makeDup(6), true}
-	jumpTable[DUP7] = jumpPtr{makeDup(7), true}
-	jumpTable[DUP8] = jumpPtr{makeDup(8), true}
-	jumpTable[DUP9] = jumpPtr{makeDup(9), true}
-	jumpTable[DUP10] = jumpPtr{makeDup(10), true}
-	jumpTable[DUP11] = jumpPtr{makeDup(11), true}
-	jumpTable[DUP12] = jumpPtr{makeDup(12), true}
-	jumpTable[DUP13] = jumpPtr{makeDup(13), true}
-	jumpTable[DUP14] = jumpPtr{makeDup(14), true}
-	jumpTable[DUP15] = jumpPtr{makeDup(15), true}
-	jumpTable[DUP16] = jumpPtr{makeDup(16), true}
-
-	jumpTable[RETURN] = jumpPtr{nil, true}
-	jumpTable[SUICIDE] = jumpPtr{nil, true}
-	jumpTable[JUMP] = jumpPtr{nil, true}
-	jumpTable[JUMPI] = jumpPtr{nil, true}
-	jumpTable[STOP] = jumpPtr{nil, true}
+var jumpTable = [256]operation{
+	ADD: operation{
+		execute:       opAdd,
+		gasCost:       makeGenericGasFunc(ADD),
```

Comment:
It feels like an additional indirection to have global gasTable and then create a function that looks up
in the table. Why not have `constGasFunc(GasFastestStep)` here? 

```go
func constGasFunc(gas *big.Int) gasFunc {
	return func(gt params.GasTable, env *Environment, contract *Contract, stack *Stack, mem *Memory, memorySize *big.Int) *big.Int {
		return gas
	}
}
```